src/apps/datasets/models.py

```diff
@@ -60,12 +63,70 @@ def save(self, *args, **kwargs):
             self.name = f"{self.created_by.username} - {self.type}"
         return super().save(*args, **kwargs)
 
+    @property
+    def in_use(self):
+        from tasks.models import Task
+        tasks = Task.objects.filter(Q(ingestion_program=self) | Q(input_data=self) | Q(reference_data=self) | Q(
+            scoring_program=self)).prefetch_related('phases')
+        phases_from_tasks = [phase for task in tasks for phase in task.phases.all()]
+        from competitions.models import Phase
+        phases = Phase.objects.filter(Q(ingestion_program=self) | Q(input_data=self) | Q(reference_data=self) | Q(
+            scoring_program=self)).prefetch_related('competition')
+        print(tasks)
+        print(phases_from_tasks)
+        print(phases)
+        task_competitions = [phase.competition.pk for phase in phases_from_tasks if phase.competition]
+        phase_competitions = [phase.competition.pk for phase in phases if phase.competition]
+        competition_set = list(set(task_competitions + phase_competitions))
+        is_used = bool(competition_set)
```

Comment:
Ditch the print statements.
Also, can simplify this logic a lot to something like this:
```
competitions = Competition.objects.filter(
    Q(phases__ingestion_program=self) |
    Q(phases__input_data=self) | 
    Q(phases__reference_data=self) | 
    Q(phases__scoring_program=self) |
    Q(phases__tasks__ingestion_program=self) |
    Q(phases__tasks__input_data=self) | 
    Q(phases__tasks__reference_data=self) | 
    Q(phases__tasks__scoring_program=self) |
).distinct()
```